doc/kuksa_requirementes.md

```diff
@@ -0,0 +1,805 @@
+   * If there is no active provider setting values while client subscription? Should Databroker or Client be aware of it?
+
+## Wildcard support
+* Should `Get` and `Subscribe` calls support Wildcard request?
+* Do we want it only for `GetMetadata`?
+
+Reference -> [Wildcard](./wildcard_matching.md)
+
+## Registration of Datapoints
+Do we need a method for providers to register data points at runtime? **Implemented in old API?**:
+
+Its purpose would be:
+1.	Publishing the availability of new data points during runtime
+2.	Claiming providership of those data points
+
+
+In a mature system some "central instance", e.g., the system integrator must be responsible to ensure that no two components can register/claim providership for any data point.
+In case of a registration method the central instance would either have to
+  * make sure there is single provider per data point via configuration, or
+  * use access rights management to enforce that only the right component can register a certain data point.
+
+
+## Availability of Datapoints
+1. The system must be capable of individually checking the availability of each data point on a specific instance of the data broker. This means verifying whether there is an active provider instance installed on the system where the data broker instance is running, which is capable of supplying the data point during the current update cycle.
+
+2. It shall be possible to determine the availability of the actual value of each data point separately on a certain instance of the data broker.
+This represents the information if the provider of that data point is up and running on the system and already provided a value of that data point.
+
+
+## Lifecycle of components
+The proper function of the overall system of components "around" the data broker, i.e., applications, providers, and the broker itself, shall not depend on a certain order of starting the components. This means:
+1.	Any clients of the data broker (applications, providers) shall not assume the availability of the data broker service when they startup.
+2.	Any clients of the data broker (applications, providers) shall not assume the permanent availability of the data broker service during their runtime.
+3.	Any applications/clients shall not assume the availability of a value for any data point at their startup.
+4.	Any applications/clients shall not assume the permanent presence of a value for any data point during their runtime.
+Explanation: Any component of the system can come and go - components could stop working due to a crash (what should not but will happen) or because of an update (which is a regular use case). Furthermore, components could reside on different execution environments which need restarts at different points of time. This shall not result in stopping and restarting the overall system. Instead, each and every component shall react in an appropriate way on changing availability of its dependencies.
```

Comment:
@lukasmittag: The reason is written in the explanation. You don't know in what order the overall collection of components is started. Also, it can happen that the broker is (temporary) not available - due to crash or update. Same is true for current signal values.
Regarding the state of signals/values:
1. A certain expected signal could be not know/available in a vehicle system, i.e. there is no respective provider installed and executable on the system (consequently, it shall not be part of the "broker config vss.json")
1. The client might have not access (rights) to it
   a. The vehicle manufacturer does not allow the client to access the signal
   b. The user ("driver") disabled access to the signal (like on smart phones: "Is this app allowed to access my camera?")
1. It might be known and accessible, but currently no value is given by a provider
1. It may have an "invalid" state, i.e. the provider is available, but maybe the physical sensor is out of order or has some temporary issues
1. The signal is known, accessible, currently provided, and has a valid value.

* The "failure" states 1 and 2(a) are "permanent". It makes no sense to install a client on the system if it needs _**mandatory**_ request to signals in those states. If the client just uses them _optionally_, it still can make sense.
* The states 3 + 4 are of temporary nature (hopefully) and consumers (and providers) need being able to proper handle these situations.
* The last state is the "ideal case" from the client's point of perspective. But the world isn't ideal ...